HYPOTHESIS TETING ON THE DIAMETER OF CULTETS

In [1]:
import pandas as pd 
import numpy as np
import scipy 
from scipy import stats
import statsmodels.api as sm
import matplotlib.pyplot as plt

HO: DIAMETER OF THE CUTLET BETWEEN TWO UNITS IS SAME ::H1=DIAMETER OF THE CUTLET BETWEEN TWO UNITS IS NOT SAME

In [2]:
# H0: data is normal,H1 data is not normal i.e givem significant lvel 5% p>0.05 H0,p<0.05 H1
data=pd.read_csv("Cutlets.csv")

In [3]:
data.head()

Unit A  Unit B
0  6.8090  6.7703
1  6.4376  7.5093
2  6.9157  6.7300
3  7.3012  6.7878
4  7.4488  7.1522

In [4]:
data_col1=data[["Unit A"]]
data_col1.head(3)

Unit A
0  6.8090
1  6.4376
2  6.9157

In [5]:
data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 35 entries, 0 to 34
Data columns (total 2 columns):
 #   Column  Non-Null Count  Dtype  
---  ------  --------------  -----  
 0   Unit A  35 non-null     float64
 1   Unit B  35 non-null     float64
dtypes: float64(2)
memory usage: 688.0 bytes


In [6]:
x1=data[["Unit A"]].mean()
x2=data[["Unit B"]].mean()
x2,x1


(Unit B    6.964297
 dtype: float64,
 Unit A    7.019091
 dtype: float64)

In [7]:
data_col2=data[["Unit B"]]
data_col2.head(3)

Unit B
0  6.7703
1  7.5093
2  6.7300

In [8]:
# twotail test and pvalue because my hypothesis assumation is type 1
value=stats.ttest_ind(data_col1,data_col2,alternative="two-sided")
value

Ttest_indResult(statistic=array([0.72286887]), pvalue=array([0.47223947]))

 p value is 0.4722 which is greater than 0.05 significant level so null Hypothesis is selected

In [9]:
var_tA=stats.normaltest(data_col1,axis=0,nan_policy='omit')
var_tA

NormaltestResult(statistic=array([2.76116448]), pvalue=array([0.25143212]))

In [10]:
var_tB=stats.normaltest(data_col2,axis=0,nan_policy='omit')
var_tB

NormaltestResult(statistic=array([1.22847999]), pvalue=array([0.54105194]))

form normalitytest if find the p value is > 0.05 so null Hypothises is chosen i.e data are normal,there is no significan differnce int he diameter of cutlets.

In [11]:
from statsmodels.stats import weightstats as stests

In [12]:
ztest,value_z= stests.ztest(x1=data["Unit A"],x2=data["Unit B"],value=0,alternative='two-sided')
print(float(value_z))
if value_z< 0.05:
    print("reject null hypothises")
else:
        print("accept null hyppothises")

0.46976045023906055
accept null hyppothises


from the ztest we come to conclusion that to accept null hypothises